app/Http/Controllers/ProjectController.php

```diff
@@ -51,16 +52,17 @@ public function index(Request $request)
      */
     public function store(Request $request)
     {
-        $project = new Project([
-          'name' => $request->get('name'),
-          'information' => $request->get('information'),
-          'deadline' => $request->get('deadline'),
-          'type' => $request->get('type'),
-          'status' => $request->get('status')
+      $request->validate(self::RULES);
+      $project = new Project([
+        'name' => $request->get('name'),
+        'information' => $request->get('information'),
+        'deadline' => $request->get('deadline'),
+        'type' => $request->get('type'),
+        'status' => $request->get('status')
         ]);
-        $project->save();
+      $project->save();
```

Comment:
You'd better use `->saveOrFail()` inside a `try..catch` block. In the `catch` block you can return a JSON error response.

Reply:
Thanks , i will fix